nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -510,12 +510,15 @@ def build_train_valid_test_datasets(self):
         max_train_steps = self.trainer.max_steps
         eval_iters = (max_train_steps // self.trainer.val_check_interval + 1) * self.trainer.limit_val_batches
         test_iters = self.trainer.limit_test_batches
-
+        
         train_valid_test_num_samples = [
             max_train_steps * global_batch_size,
             eval_iters * global_batch_size,
             test_iters * global_batch_size,
         ]
+        if self.trainer.limit_val_batches == 1.0 and type(self.trainer.limit_val_batches) == float:
```

Comment:
Can you raise an error here if `limit_val_batches` is not an int or != `1.0`? I think intermediate values like `0.6` etc do not work as expected right?

Reply:
Yes. Sure will do that

Reply:
Actually I think it will work. If the limit val batches is a float value less than 1, it will work as long as we set the train_valid_test_num_samples[1] to 1. . I am going to == 1.0 to <= 1.0 condition. I tested it out on my local box. 
I will raise an error if the limit val batches is set to a value greater than 1.0.